### PR TITLE
Add DatanoiseTV DSP Board.

### PR DESCRIPTION
**Pull request by DatanoiseTV**
Adds the DatanoiseTV RP2040 Audio DSP board.
https://twitter.com/DatanoiseTV/status/1536705299964481540

### CODE REVIEW

File: src/boards/include/boards/datanoisetv_rp2040_dsp.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Raspberry Pi (Trading) Ltd.
+ *
```
Comment (JamesH65) [inline src/boards/include/boards/datanoisetv_rp2040_dsp.h:3]: I suspect this copyright is wrong.

**Comment by JamesH65** [comment]
Is this product actually for sale at this point?

**Comment by DatanoiseTV** [comment]
> Is this product actually for sale at this point?

It is available on GitHub as an open source project, with BOM and assembly files for JLCPCB assembly.
I am planning to keep it as an open source DIY project.

File: src/boards/include/boards/datanoisetv_rp2040_dsp.h (hunk 1 above)
Comment (DatanoiseTV) [reply src/boards/include/boards/datanoisetv_rp2040_dsp.h:3]: I have copied it from the other board files, which also didn't seem to change the copyright.

File: src/boards/include/boards/datanoisetv_rp2040_dsp.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+// --- UART ---
+#ifndef PICO_DEFAULT_UART
+#define PICO_DEFAULT_UART 0
+#endif
+#ifndef PICO_DEFAULT_UART_TX_PIN
+#define PICO_DEFAULT_UART_TX_PIN 0
+#endif
+#ifndef PICO_DEFAULT_UART_RX_PIN
+#define PICO_DEFAULT_UART_RX_PIN 1
+#endif
```
Comment (lurch) [inline src/boards/include/boards/datanoisetv_rp2040_dsp.h:32]: These are the same as the values in `pico.h`, which you're `#include`-ing at the bottom of this file anyway?

Comment (DatanoiseTV) [reply src/boards/include/boards/datanoisetv_rp2040_dsp.h:32]: Correct, you can delete these.

Comment (lurch) [reply src/boards/include/boards/datanoisetv_rp2040_dsp.h:32]: This is _your_ PR, not mine :wink:

Comment (DatanoiseTV) [reply src/boards/include/boards/datanoisetv_rp2040_dsp.h]: Done.

**Review by kilograham — APPROVED** [review]

**Pull request merged**